### PR TITLE
Upgrade constant column feature to support remove & replace column and also resolve any bugs

### LINKED ISSUE #267: Migration (and Validation) fails when constant column feature adds PK columns with different data-types than existing PK columns
To replicate the issue do the below steps

Create the below `origin` and `target` tables
```
CREATE TABLE origin_constant_column(key text, value text, PRIMARY KEY (key));
INSERT INTO origin_constant_column(key,value) VALUES ('key1','valueA');
INSERT INTO origin_constant_column(key,value) VALUES ('key2','valueB');
INSERT INTO origin_constant_column(key,value) VALUES ('key3','valueC');

CREATE TABLE target_constant_column(const1 int, key text, value text, const2 int, PRIMARY KEY (const1, key));
```

Add below lines to the config file
```
spark.cdm.feature.constantColumns.names        const1,const2
spark.cdm.feature.constantColumns.types        int,int
spark.cdm.feature.constantColumns.values       1,2
```

Run migration (or validation) job as below
`spark-submit --properties-file cdm.properties --conf spark.cdm.schema.origin.keyspaceTable="test_ks.origin_constant_column" --conf spark.cdm.schema.target.keyspaceTable="test_ks.target_constant_column" --master "local"  --class com.datastax.cdm.job.Migrate cassandra-data-migrator-4.1.16.jar`

This throws the below exception
```
24/07/01 10:36:24 INFO CopyJobSession: CQL -- origin select: SELECT key,value,TTL(value),WRITETIME(value) FROM test_ks.origin_constant_column WHERE TOKEN(key) >= ? AND TOKEN(key) <= ? ALLOW FILTERING
24/07/01 10:36:24 INFO CopyJobSession: CQL -- target select: SELECT const1,key,const2,value FROM test_ks.target_constant_column WHERE const1=1 AND key=?
24/07/01 10:36:24 INFO CopyJobSession: CQL -- target upsert: INSERT INTO test_ks.target_constant_column (key,value,const1,const2) VALUES (?,?,1,2) USING TTL ? AND TIMESTAMP ?
24/07/01 10:36:24 INFO CopyJobSession: ThreadID: 106 Processing min: -9223372036854775808 max: 9223372036854775807
24/07/01 10:36:24 ERROR CopyJobSession: Error occurred during Attempt#: 1
com.datastax.oss.driver.api.core.type.codec.CodecNotFoundException: Codec not found for requested operation: [INT <-> java.lang.String]
	at com.datastax.oss.driver.internal.core.type.codec.registry.CachingCodecRegistry.createCodec(CachingCodecRegistry.java:656)
	at com.datastax.oss.driver.internal.core.type.codec.registry.DefaultCodecRegistry$1.load(DefaultCodecRegistry.java:95)
	at com.datastax.oss.driver.internal.core.type.codec.registry.DefaultCodecRegistry$1.load(DefaultCodecRegistry.java:92)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2276)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2154)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache$Segment.get(LocalCache.java:2044)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache.get(LocalCache.java:3951)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache.getOrLoad(LocalCache.java:3973)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4957)
	at com.datastax.oss.driver.shaded.guava.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4963)
	at com.datastax.oss.driver.internal.core.type.codec.registry.DefaultCodecRegistry.getCachedCodec(DefaultCodecRegistry.java:117)
	at com.datastax.oss.driver.internal.core.type.codec.registry.CachingCodecRegistry.codecFor(CachingCodecRegistry.java:209)
	at com.datastax.cdm.data.CqlConversion.convert_CODEC(CqlConversion.java:246)
	at com.datastax.cdm.data.CqlConversion.convert_ONE(CqlConversion.java:227)
	at com.datastax.cdm.data.CqlConversion.convert(CqlConversion.java:99)
	at com.datastax.cdm.schema.CqlTable.getAndConvertData(CqlTable.java:305)
	at com.datastax.cdm.data.PKFactory.getTargetPKValuesFromOriginColumnLookupMethod(PKFactory.java:241)
	at com.datastax.cdm.data.PKFactory.getTargetPK(PKFactory.java:98)
	at com.datastax.cdm.job.CopyJobSession.getDataAndInsert(CopyJobSession.java:86)
	at com.datastax.cdm.job.CopyJobSession.processSlice(CopyJobSession.java:62)
	at com.datastax.cdm.job.CopyJobSession.processSlice(CopyJobSession.java:36)
	at com.datastax.cdm.job.Migrate$.$anonfun$execute$3(Migrate.scala:28)
	at com.datastax.cdm.job.Migrate$.$anonfun$execute$3$adapted(Migrate.scala:26)
	at com.datastax.spark.connector.cql.CassandraConnector.$anonfun$withSessionDo$1(CassandraConnector.scala:104)
	at com.datastax.spark.connector.cql.CassandraConnector.closeResourceAfterUse(CassandraConnector.scala:121)
	at com.datastax.spark.connector.cql.CassandraConnector.withSessionDo(CassandraConnector.scala:103)
	at com.datastax.cdm.job.Migrate$.$anonfun$execute$2(Migrate.scala:26)
	at com.datastax.cdm.job.Migrate$.$anonfun$execute$2$adapted(Migrate.scala:25)
	at com.datastax.spark.connector.cql.CassandraConnector.$anonfun$withSessionDo$1(CassandraConnector.scala:104)
	at com.datastax.spark.connector.cql.CassandraConnector.closeResourceAfterUse(CassandraConnector.scala:121)
	at com.datastax.spark.connector.cql.CassandraConnector.withSessionDo(CassandraConnector.scala:103)
	at com.datastax.cdm.job.Migrate$.$anonfun$execute$1(Migrate.scala:25)
	at com.datastax.cdm.job.Migrate$.$anonfun$execute$1$adapted(Migrate.scala:24)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
	at org.apache.spark.InterruptibleIterator.foreach(InterruptibleIterator.scala:28)
	at org.apache.spark.rdd.RDD.$anonfun$foreach$2(RDD.scala:1031)
	at org.apache.spark.rdd.RDD.$anonfun$foreach$2$adapted(RDD.scala:1031)
	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2438)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
	at org.apache.spark.scheduler.Task.run(Task.scala:141)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
24/07/01 10:36:24 ERROR CopyJobSession: Error with PartitionRange -- ThreadID: 106 Processing min: -9223372036854775808 max: 9223372036854775807 -- Attempt# 1
24/07/01 10:36:24 ERROR CopyJobSession: Error stats READ=1, SKIPPED=0, UNFLUSHED=0, WRITE=0, ERROR=1
24/07/01 10:36:24 INFO Executor: Finished task 0.0 in stage 0.0 (TID 0). 981 bytes result sent to driver
```

As you can infer from the above exception, this is because the target table now has a PK that starts with an INT datatype, whereas the origin PK starts with String datatype. The root cause is because CDM is not correctly able to map constant columns with PK changes from origin to target.

### PR DESCRIPTION
**Pull request by pravinbhat**
**What this PR does**:
Upgrades the `constant column` feature to support `remove` & `replace` of columns from `origin` to `target` and also resolve any bugs

**Which issue(s) this PR fixes**:
Fixes #267 

**Checklist:**
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by msmygit — CHANGES_REQUESTED** [review]
CI failures [here](https://github.com/datastax/cassandra-data-migrator/actions/runs/9863605136/job/27236744094?pr=268#step:4:34247) (though values looks correct to me)

**Comment by pravinbhat** [comment]
> CI failures [here](https://github.com/datastax/cassandra-data-migrator/actions/runs/9863605136/job/27236744094?pr=268#step:4:34247) (though values looks correct to me)

Fixed!

File: RELEASE.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # Release Notes
+## [4.2.0] - 2024-07-09
```
Comment (msmygit) [inline RELEASE.md:2]: I see that we're bumping to `4.2` now. Does this have any backward incompatibility changes and hence the need to bump the minor?

**Review by msmygit — APPROVED** [review]
lgtm 👍🏼

File: RELEASE.md (hunk 1 above)
Comment (pravinbhat) [reply RELEASE.md:2]: No, as this version adds an important feature (customer ask), we are bumping the minor version. We follow [semantic versioning](https://semver.org/).

**Review by mfmaher2 — APPROVED** [review]

**Pull request merged**